sw/device/tests/sysrst_ctrl_in_irq_test.c

```diff
@@ -47,25 +59,61 @@ static const dif_pinmux_index_t kPeripheralInputs[] = {
     kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonLidOpen,
 };
 
-static const dif_pinmux_index_t kInputPads[] = {
+static const dif_pinmux_index_t kInputPadsDV[] = {
     kTopEarlgreyPinmuxInselIob3, kTopEarlgreyPinmuxInselIob6,
     kTopEarlgreyPinmuxInselIob8, kTopEarlgreyPinmuxInselIor13,
     kTopEarlgreyPinmuxInselIoc7, kTopEarlgreyPinmuxInselIoc9,
 };
 
+// We need different pins on the hyperdebug boards since certain
+// pins are not routed to the hyperdebug.
+static const dif_pinmux_index_t kInputPadsReal[] = {
+    kTopEarlgreyPinmuxInselIor10, kTopEarlgreyPinmuxInselIor11,
+    kTopEarlgreyPinmuxInselIor12, kTopEarlgreyPinmuxInselIor5,
+    kTopEarlgreyPinmuxInselIor6,  kTopEarlgreyPinmuxInselIor7,
+};
+
 void test_phase_sync(void) {
   test_status_set(kTestStatusInTest);
   test_status_set(kTestStatusInWfi);
 }
 
+static void wait_for_any_interrupt(void) {
+  // Disable interrupts to be certain interrupt doesn't occur between while
+  // condition check and `wait_for_interrupt` (so WFI misses that interrupt).
+  irq_global_ctrl(false);
+
+  // Only enter WFI loop if we haven't already seen the interrupt.
+  while (peripheral == UINT32_MAX) {
+    wait_for_interrupt();
+    // WFI ignores global interrupt enable, so enable it now and then
```

Comment:
The WFI instruction is funny, since it is meant to ignore mstatus *ie bits, so it is not clear what this code accomplishes.
 I guess the intent is to continue to wait for interrupts until we get the one that sets peripheral? But then why bother calling irq_global_ctrl?

Reply:
This is to guard against the case where the interrupt was called before we wait for an interrupt: in this case the wait for interrupt will block forever. This means we have to check for the value of `peripheral` but we need to do so atomically because the interrupt could happen between the check and the call to wfi. The point of calling irq_global_ctrl is to control where the interrupt is called so that we can atomically check the `peripheral` variable. Hopefully this makes sense?

Reply:
You explanation here is clearer than the comments. Perhaps the comments could be improved?

Reply:
Thanks @matutem , I will change the comment.

Reply:
I have updated the comment, hopefully this is better.